examples/silentpayments.c

```diff
@@ -0,0 +1,417 @@
+         *
+         * Alice creates the recipient objects and adds the index of the original ordering (i.e. the ordering
+         * of the `sp_addresses` array) to each object. This index is used to return the generated outputs in
+         * the original ordering so that Alice can match up the generated outputs with the correct amounts.
+         *
+         * Note: to create multiple outputs for Carol, Alice simply passes her silent payment
+         * address mutltiple times.
+         */
+        sp_addresses[0] = &carol_address; /* : 1.0 BTC */
+        sp_addresses[1] = &bob_address;   /* : 2.0 BTC */
+        sp_addresses[2] = &carol_address; /* : 3.0 BTC */
+        for (i = 0; i < N_TX_OUTPUTS; i++) {
+            ret = secp256k1_ec_pubkey_parse(ctx, &recipients[i].scan_pubkey, (*(sp_addresses[i]))[0], 33);
+            assert(ret);
+            ret = secp256k1_ec_pubkey_parse(ctx, &recipients[i].spend_pubkey, (*(sp_addresses[i]))[1], 33);
+            assert(ret);
+            recipients[i].index = i;
+            recipient_ptrs[i] = &recipients[i];
+        }
+        for (i = 0; i < N_TX_OUTPUTS; i++) {
+            generated_output_ptrs[i] = &tx_outputs[i];
+        }
+        ret = secp256k1_silentpayments_sender_create_outputs(ctx,
+            generated_output_ptrs,
+            recipient_ptrs, N_TX_OUTPUTS,
+            smallest_outpoint,
+            sender_seckey_ptrs, N_TX_INPUTS,
+            NULL, 0
+        );
+        assert(ret);
+        printf("Alice created the following outputs for Bob and Carol: \n");
+        for (i = 0; i < N_TX_OUTPUTS; i++) {
+            printf("    ");
+            printf("%s : ", address_amounts[i]);
+            secp256k1_xonly_pubkey_serialize(ctx, xonly_print, &tx_outputs[i]);
+            print_hex(xonly_print, sizeof(xonly_print));
+        }
+        /* It's best practice to try to clear secrets from memory after using them.
+         * This is done because some bugs can allow an attacker to leak memory, for
+         * example through "out of bounds" array access (see Heartbleed), Or the OS
+         * swapping them to disk. Hence, we overwrite the secret key buffer with zeros.
+         *
+         * Here we are preventing these writes from being optimized out, as any good compiler
+         * will remove any writes that aren't used. */
+        for (i = 0; i < N_TX_INPUTS; i++) {
+            secure_erase(&sender_seckeys[i], sizeof(sender_seckeys[i]));
+        }
+    }
+
+    /*** Receiving ***/
+    {
+        /*** Transaction data ***
+         *
+         * Here we create a few global variables to represent the transaction data:
+         *
+         *     1. The transaction inputs, `tx_input_ptrs`
+         *     2. The transaction outputs, `tx_output_ptrs`
+         *
+         * These will be used to demonstrate scanning as a full node and scanning as a light client.
+         */
+        const secp256k1_xonly_pubkey *tx_input_ptrs[N_TX_INPUTS];
```

Comment:
c71791c629ec88e2e253aae4157d7113b60bee60

```c
/** For each Silent Payment eligible input we collect all pubkeys
  * found in the scriptSig or witness. For simplicity in this 
  * example inputs have only one pubkey.
  */
const secp256k1_xonly_pubkey *tx_input_pubkey_ptrs[N_TX_INPUTS];
```

Reply:
Actually, the eligible inputs from the BIP were chosen so that there is only one public key per input. This is why `OP_CHECKMULTISIG` inputs are not eligible for silent payments, for example.

Reply:
Ah, nice! In that case:

/** For each Silent Payment eligible input we collect the pubkey
  * found in the scriptSig or witness. See Inputs For Shared Secret Derivation
  * in the BIP.
  */

Reply:
This comment still feels like protocol exposition and doesn't add any information on how to use the API, which is what the example is aimed at, so I'd prefer to keep it as is.